api/app/Repositories/Criteria/SearchByNameStringCriteria.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repositories\Criteria;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Facades\DB;
+
+final class SearchByNameStringCriteria implements Criteria
+{
+    private ?string $searchString;
+
+    public function __construct(?string $searchString)
+    {
+        $this->searchString = $searchString ? mb_strtolower($searchString) : '';
+    }
+
+    public function apply(Builder $builder): Builder
+    {
+        return $builder->where(
+            DB::raw('LOWER(name)'),
+            'like',
+            $this->searchString . '%'
+        );
```

Comment:
I am not sure we need to transform `name` to lowercase as we use COLLATION that allows case insensitive search. Also, the rule must be `%string%` as we search not only from the beginning of the name.

Reply:
It will not be work, without 'LOWER', i've tried.